code/controllers/subsystem/gamemaster.dm

```diff
@@ -0,0 +1,261 @@
+	spawn(50)
+		gamemaster_announce("Why are you so nervous?")
+
+/datum/subsystem/gamemaster/proc/singuloth()
+	var/turf/T = get_turf(safepick(generic_event_spawns))
+	if(!T)
+		var/area/A = safepick(z1_areas)
+		T = get_turf(safepick(A.contents))
+
+	gamemaster_announce("The engineering department has started a singularity at [T.loc].")
+	new /obj/singularity(T)
+
+/datum/subsystem/gamemaster/proc/el_dorado()
+	var/area/A = safepick(z1_areas)
+	if(A)
+		for(var/turf/T in A)
+			if(T.density)
+				T.ChangeTurf(/turf/simulated/wall/mineral/gold)
+			else
+				T.ChangeTurf(/turf/simulated/floor/mineral/gold)
+			CHECK_TICK
+		gamemaster_announce("The wonders of space alchemy have transformed [A] forever.")
+
+/datum/subsystem/gamemaster/proc/el_honkado()
+	var/area/A = safepick(z1_areas)
+	if(A)
+		for(var/turf/T in A)
+			if(T.density)
+				T.ChangeTurf(/turf/simulated/wall/mineral/clown)
+			else
+				T.ChangeTurf(/turf/simulated/floor/mineral/bananium)
+			CHECK_TICK
+		gamemaster_announce("The wonders of clown alchemy have transformed [A] forever.")
+
+/datum/subsystem/gamemaster/proc/privacy_windows()
+	for(var/V in z1_areas)
+		var/area/A = V
+		for(var/obj/structure/window/W in A)
+			W.color = "#101010"
+			W.opacity = 1
+			CHECK_TICK
+	gamemaster_announce("I turned all the windows opaque for enhanced privacy.")
+
+/datum/subsystem/gamemaster/proc/give_guns()
+	for(var/V in player_list)
+		var/mob/M = V
+		M.put_in_hands(new /obj/item/weapon/gun/projectile/automatic/pistol/deagle(M))
+	gamemaster_announce("An armed crew, is a civilized crew.")
+
+/datum/subsystem/gamemaster/proc/ai_cameras()
+	for(var/V in z1_areas)
+		var/area/A = V
+		for(var/obj/machinery/camera/C in A)
+			C.icon = 'icons/mob/AI.dmi'
+			C.icon_state = "ai-holo-old"
+	gamemaster_announce("Remember crew, the AI is ever vigilant.")
+
+/datum/subsystem/gamemaster/proc/communism()
+	for(var/V in z1_areas)
+		var/area/A = V
+		for(var/obj/machinery/door/door in A)
+			door.req_access_txt = ""
+			door.req_one_access_txt = ""
+			door.req_access = null
+			door.req_one_access = null
+			CHECK_TICK
+		for(var/obj/structure/closet/closet in A)
+			closet.req_access_txt = ""
+			closet.req_one_access_txt = ""
+			closet.req_access = null
+			closet.req_one_access = null
+			CHECK_TICK
+	gamemaster_announce("All crew members are equal! VIVA!")
+
+/datum/subsystem/gamemaster/proc/knockdown()
+	gamemaster_announce("Buckle up!")
+	spawn(50)
+		for(var/mob/living/carbon/M in player_list)
+			if(M.buckled)
+				shake_camera(M, 2, 1) // turn it down a bit come on
+			else
+				shake_camera(M, 7, 1)
+				if(istype(M))
+					M.Weaken(3)
+
+/datum/subsystem/gamemaster/proc/change_dir()
+	var/direction = pick(SOUTH, EAST, WEST)
+	for(var/client/C in clients)
+		if(!(C in admins))
+			C.dir = direction
+	gamemaster_announce("This will take some time for you to get used to...")
+	spawn(3600)
+		for(var/client/C in clients)
+			C.dir = NORTH
```

Comment:
Darnit.  Why has everyone been wanting to screw with client.dir lately?  Here's your filthy icon smoothing fix. https://gist.github.com/YotaXP/4b996eb96f4a39b0ea0a

*Does not fix bar sign.
*Does not fix "Space Station 13" logo on floor.
*Does not fix derpy maps where some turfs have dir=0. (I'm looking at you, tiles in front of medbay on Box)